edb/schema/name.py

```diff
@@ -179,15 +179,20 @@ def mangle_name(name: str) -> str:
     )
 
 
+mangle_re_1 = re.compile(r'(?<![|])\|(?![|])')
+mangle_re_2 = re.compile(r'(?<![&])&(?![&])')
+
+
 def unmangle_name(name: str) -> str:
-    name = re.sub(r'(?<![|])\|(?![|])', '::', name)
-    name = re.sub(r'(?<![&])&(?![&])', '@', name)
+    name = mangle_re_1.sub('::', name)
```

Comment:
Does this func need an lru_cache?

Reply:
No, it's called by another function that is already cached.